### ISSUE TITLE
Federated (client-side) evaluation omitted if there is one configured server-side. Flower 0.17.0

### ISSUE THREAD

**Issue by xpetrov** [adap/flower#913]
Hi, in my project, I would like to evaluate the model both client-side and server-side. As I see, the framework ignores federated evaluation if there is centralized evaluation configured. See `\flwr\server\strategy\fedavg.py:210`

```
def configure_evaluate(
    self, rnd: int, parameters: Parameters, client_manager: ClientManager
) -> List[Tuple[ClientProxy, EvaluateIns]]:
    """Configure the next round of evaluation."""
    # Do not configure federated evaluation if a centralized evaluation
    # function is provided
    if self.eval_fn is not None:
        return []
    ...
    ...
```

Is there any reason for such behavior?

**Comment by danieljanes** [comment]
Hi @xpetrov , thanks for asking! This is a usability artifact from the very early days of Flower, and to be honest, I don't think it makes sense any more. Changing it would mean a breaking change, but we have a few of those ahead of us anyways, so I'd say let's fix it.

There's room for different solutions here, and any solution will need to enable users to selectively enable/disable both centralized evaluation and federated  evaluation. One way to do it would be to make it implicit: if the user initialized the strategy with a `fraction_eval > 0.0`, then select clients and perform federated eval. If `fraction_eval == 0.0` (or `None`), then disable federated eval. For centralized eval, we'd make it dependent on whether an evaluation function has been provided or not. And, for the sake of completeness, when the user initializes the strategy without passing neither `fraction_eval` nor an evaluation function, we'd set some default value for `fraction_eval` that's `>0.0`.

@xpetrov would this work for you?

CC'ing @tanertopal @pedropgusmao

**Comment by xpetrov** [comment]
Thank you @danieljanes for your answer. Sure, looks to me as a good approach.

Looking forward to future releases

**Comment by danieljanes** [comment]
Closing this issue, the feature has been released with Flower 0.19, thanks again for bringing it up @xpetrov!
